sdk/core/azure-core-http-vertx/src/test/java/com/azure/core/http/vertx/implementation/VertxRequestWriteSubscriberTests.java

```diff
@@ -0,0 +1,251 @@
+            sink.complete();
+        });
+        Buffer writtenContent = Buffer.buffer();
+        HttpClientRequest httpClientRequest = new MockHttpClientRequest() {
+            @Override
+            public void write(Buffer data, Handler<AsyncResult<Void>> handler) {
+                writtenContent.appendBuffer(data);
+                Promise<Void> promise = Promise.promise();
+                promise.future().onComplete(handler);
+                promise.complete();
+            }
+
+            @Override
+            public Future<Void> end() {
+                promise.complete();
+                return super.end();
+            }
+        };
+
+        VertxRequestWriteSubscriber subscriber
+            = new VertxRequestWriteSubscriber(httpClientRequest, promise, null, Context.empty());
+
+        data.subscribe(subscriber);
+
+        waitForCompletion(promise);
+        assertTrue(promise.future().succeeded(), "The request should have succeeded.");
+        assertArraysEqual(DATA, writtenContent.getBytes());
+    }
+
+    /**
+     * Tests that request writing operates as expected when the reactive stream has onNext followed by onComplete.
+     * <p>
+     * The result of this operation order should be that the request is failed and there are no errors dropped. While
+     * the reactive stream completed successfully the writing of the request failed.
+     */
+    @Test
+    public void onNextWithWriteFailureThenOnComplete() throws InterruptedException {
+        RuntimeException writeError = new RuntimeException("Write error");
+        AtomicReference<Throwable> onErrorDropped = new AtomicReference<>();
+        Consumer<Throwable> onErrorDroppedConsumer = onErrorDropped::set;
+        Promise<HttpResponse> promise = Promise.promise();
+        Flux<ByteBuffer> data = Flux.generate(sink -> {
+            sink.next(ByteBuffer.wrap(DATA));
+            sink.complete();
+        });
+        HttpClientRequest httpClientRequest = new MockHttpClientRequest() {
+            @Override
+            public void write(Buffer data, Handler<AsyncResult<Void>> handler) {
+                Promise<Void> promise = Promise.promise();
+                promise.future().onComplete(handler);
+                promise.fail(writeError);
+            }
+        };
+
+        VertxRequestWriteSubscriber subscriber = new VertxRequestWriteSubscriber(httpClientRequest, promise, null,
+            Context.of("reactor.onErrorDropped.local", onErrorDroppedConsumer));
+
+        data.subscribe(subscriber);
+
+        waitForCompletion(promise);
+        assertTrue(promise.future().failed(), "The request should have failed.");
+        assertSame(writeError, promise.future().cause(), "The write error should be the cause of the failure.");
+        assertNull(onErrorDropped.get(), "No reactive error should be dropped.");
+    }
+
+    /**
+     * Tests that if the promise being used to manage the request-response state succeeds externally while writing the
```

Comment:
'Ensures that the promise managing the request-response state is appropriately handled when it succeeds externally during the request writing process.'